system/Config/BaseConfig.php

```diff
@@ -61,23 +83,25 @@ public function __construct()
     {
         static::$moduleConfig = config(Modules::class);
 
-        $this->registerProperties();
-
-        $properties  = array_keys(get_object_vars($this));
-        $prefix      = static::class;
-        $slashAt     = strrpos($prefix, '\\');
-        $shortPrefix = strtolower(substr($prefix, $slashAt === false ? 0 : $slashAt + 1));
-
-        foreach ($properties as $property) {
-            $this->initEnvValue($this->{$property}, $property, $prefix, $shortPrefix);
-
-            if ($this instanceof Encryption && $property === 'key') {
-                if (strpos($this->{$property}, 'hex2bin:') === 0) {
-                    // Handle hex2bin prefix
-                    $this->{$property} = hex2bin(substr($this->{$property}, 8));
-                } elseif (strpos($this->{$property}, 'base64:') === 0) {
-                    // Handle base64 prefix
-                    $this->{$property} = base64_decode(substr($this->{$property}, 7), true);
+        if (static::$override) {
```

Comment:
I prefer negation check to have less deep code, eg:

```php
if (! static::$override) {
    return;
}

$this->registerProperties();
```

the git diff also should have less noise

Reply:
Exactly. Refactored.